### PR TITLE
Feature/mparker/mindbody get queries

### PR DESCRIPTION
**Pull request by MadParkD**
Contains all get requests for the mindbody api.
To see the function decorations and parameters in vscode use a jsdoc style @type annotation to declare where it should look
```javascript
import mindbodyQueries from "../../api-manager.js";
/**
 * @type mindbodyQueries:MindbodyQueries
 */
var minbodyAPI = new MindbodyQueries();
```
mindbodyAPI.getSomething should autocomplete

### CODE REVIEW

File: server/api-manager.js [hunk 1]
```diff
@@ -35,32 +39,895 @@ class MindbodyQueries {
...
       APIKEY,
       SITEID,
       "GET",
       ""
     );
-    req.addAuth(this.authToken);
+    return this.decorateAndMake(request);
+  }
+
+  /* eslint-disable *//**
```
Comment (DanielBrewerPsu) [inline server/api-manager.js:59]: This isn't a deal breaker but it seems silly to use eslint to enforce some code standards but then eslint-disable them all the time. I'd rather:
 - change our eslint settings to be compatible with the kinds of comments we write
 - or find a way to write comments that are compatible with our eslint settings.

File: server/api-manager.js [hunk 2]
```diff
@@ -35,32 +39,895 @@ class MindbodyQueries {
...
-      return req.makeRequest();
+      return request.makeRequest();
     }
     return Promise.reject(Error("Request limit reached"));
   }
 
+
   //may be changed later
   atLimit() {
+    /*eslint-disable no-magic-numbers */
```
Comment (DanielBrewerPsu) [inline server/api-manager.js:928]: I think it's better to give eslist what it wants here and write like:
    const MAX_REQUESTS_DESIRED = 800;
    if (this.requestNum >= MAX_REQUESTS_DESIRED) {
      return true;

File: server/api-manager.js [hunk 3]
```diff
@@ -35,32 +39,895 @@ class MindbodyQueries {
...
+    return this.decorateAndMake(request);
+  }
+
+  /* eslint-disable *//**
+  * Required:
+  *  @param SessionTypeId 
+  * Optional:
+  *  @param Filters  
+  *  @param StaffIds  
+  *  @param StaffIds
```
Comment (DanielBrewerPsu) [inline server/api-manager.js:882]: doubled

File: server/api-manager.js (hunk 1 above)
Comment (MadParkD) [reply server/api-manager.js:59]: Eslint is fairly strict and I am using it to auto-format as well. No amount of me tweaking the settings seemed to really let me put something easily human readable in while maintaining the tooltip formatting, so I went with this. I am open to doing this with some eslint changes, but I was having a little trouble getting it to work.

File: server/api-manager.js (hunk 3 above)
Comment (MadParkD) [reply server/api-manager.js]: Good catch

File: server/api-manager.js (hunk 2 above)
Comment (MadParkD) [reply server/api-manager.js]: Agreed, btw you can suggest changes with a dropdown when you are in review mode, which lets the owner push a button to merge the change.

**Pull request merged**